src/store/LastDigitsStore.js

```diff
@@ -0,0 +1,80 @@
+import { observable, action, computed, reaction } from 'mobx';
+
+export default class LastDigitsStore {
+    constructor(mainStore) {
+        this.mainStore = mainStore;
+        reaction(() => this.mainStore.state.showLastDigitStats, this.showLastDigitStats);
+    }
+
+    get context() { return this.mainStore.chart.context; }
+    get stx() { return this.context.stx; }
+
+    minHeight = 40;
+    maxHeight = 100;
+    gradiantLine = this.maxHeight / 2;
+    digits = [];
+    latestData = [];
+    @observable bars = [];
+
+    get api() {
+        return this.mainStore.chart.api;
+    }
+
+    @computed get decimalPlaces() {
+        return this.mainStore.chart.currentActiveSymbol.decimal_places;
+    }
+
+    @computed get isVisible() {
+        return this.mainStore.state.showLastDigitStats;
+    }
+
+    @computed get marketDisplayName() {
+        return this.mainStore.chart.currentActiveSymbol ? this.mainStore.chart.currentActiveSymbol.name : '';
+    }
+
+    @action.bound async showLastDigitStats() {
+        if (!this.context) return;
+        if (this.mainStore.state.showLastDigitStats) {
+            for (let i = 0; i < 10; i++) {
+                this.digits.push(0);
+                this.bars.push({ height:0, cName:'' });
+            }
+            const tickHistory = await this.api.getTickHistory({ symbol :this.mainStore.chart.currentActiveSymbol.symbol });
```

Comment:
@MahboobehMohammadi I'd be better to define the `1000` in the request here instead of using `BinaryAPI.DEFAULT_COUNT`. If in the future because of any reason the default value is changed, it won't affect the functionality of the component.

Indeed, we can use the previous response of `ticks_history` instead of making another request. If they are not as many as 1000 ticks then we can send a request to get the latest 1000.